File: leetcode/q90_sprint/arrays_and_hashing/two_sum/Solution.py
from typing import List, Dict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Time Complexity O(n)
        # Space Complexity O(n)

        ans: List[int] = [-1, -1]
        seen: Dict[int, int] = dict()

        for index, elem in enumerate(nums): 
            complement: int = target - elem
            if (complement in seen):
                ans[0] = seen[complement]
                ans[1] = index
                break

            else:
                seen[elem] = index

        return ans

File: leetcode/q90_sprint/arrays_and_hashing/two_sum/Solution.java
package q90_sprint.arrays_and_hashing.two_sum;

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Time Complexity O(n)
        // Space Complexity O(n)
        Map<Integer, Integer> hash_map = new HashMap<>();
        int[] ans = {-1, -1};

        for (int i = 0; i < nums.length; ++i) {
            int complement = target - nums[i];
            if (hash_map.containsKey(complement)) {
                ans[0] = hash_map.get(complement);
                ans[1] = i;
                break;
            }
            else {
                hash_map.put(nums[i], i);
            }
        }    

        return ans;
    }
}